### PR TITLE
use mfree to free memory and set NULL

### PR DESCRIPTION
**Pull request by weizhixiangcoder**

### CODE REVIEW

File: src/libsystemd-network/test-dhcp6-client.c [hunk 1]
```diff
@@ -148,6 +150,9 @@ static void test_parse_domain(void) {
         _cleanup_free_ char *domain = NULL;
         _cleanup_strv_free_ char **list = NULL;
         uint8_t *data;
+        char *domain = NULL;
+        char **list = NULL;
```
Comment (yuwata) [inline src/libsystemd-network/test-dhcp6-client.c]: These shadows the above declarations.

File: src/libsystemd-network/test-dhcp6-client.c [hunk 2]
```diff
@@ -148,6 +150,9 @@ static void test_parse_domain(void) {
...
+        int r;
```
Comment (yuwata) [inline src/libsystemd-network/test-dhcp6-client.c]: This is not used.

File: src/libsystemd-network/test-dhcp6-client.c [hunk 3]
```diff
@@ -25,6 +25,8 @@
 #include "strv.h"
 #include "tests.h"
 #include "time-util.h"
+#include "virt.h"
+#include "alloc-util.h"
```
Comment (yuwata) [inline src/libsystemd-network/test-dhcp6-client.c]: These are not necessary, and not sorted alphabetically.

File: src/test/test-cgroup-util.c [hunk 4]
```diff
@@ -392,18 +393,18 @@ TEST(cg_get_keyed_attribute) {
 
         assert_se(cg_get_keyed_attribute("cpu", "/init.scope", "cpu.stat", STRV_MAKE("usage_usec", "no_such_attr"), vals3) == -ENXIO);
         assert_se(cg_get_keyed_attribute_graceful("cpu", "/init.scope", "cpu.stat", STRV_MAKE("usage_usec", "no_such_attr"), vals3) == 1);
-        assert_se(vals3[0] && !vals3[1]);
-        free(vals3[0]);
+        assert(vals3[0] && !vals3[1]);
```
Comment (yuwata) [inline src/test/test-cgroup-util.c]: Why do you convert `assert_se()` into `assert()`? `assert()` may be disabled when -DNDEBUG is provided to the compiler command line option. Moreover this is a testing code, please use `assert_se()`.

**Review by yuwata — CHANGES_REQUESTED** [review]
Please use `git push --force` when you update your existing PR, then it is not necessary to recreate a new PR.

**Comment by weizhixiangcoder** [comment]
> 

OK

File: src/libsystemd-network/test-dhcp6-client.c [hunk 5]
```diff
@@ -13,6 +13,7 @@
 #include "sd-dhcp6-client.h"
 #include "sd-event.h"
 
+#include "alloc-util.h"
```
Comment (yuwata) [inline src/libsystemd-network/test-dhcp6-client.c]: I cannot catch why you add the header. Do you have any issue on the test?

File: src/test/test-cgroup-util.c [hunk 6]
```diff
@@ -393,17 +393,17 @@ TEST(cg_get_keyed_attribute) {
...
         assert_se(cg_get_keyed_attribute_graceful("cpu", "/init.scope", "cpu.stat", STRV_MAKE("usage_usec", "no_such_attr"), vals3) == 1);
         assert_se(vals3[0] && !vals3[1]);
-        free(vals3[0]);
+        vals3[0] = mfree(vals3[0]);
 
         assert_se(cg_get_keyed_attribute("cpu", "/init.scope", "cpu.stat", STRV_MAKE("usage_usec", "usage_usec"), vals3) == -ENXIO);
         assert_se(cg_get_keyed_attribute_graceful("cpu", "/init.scope", "cpu.stat", STRV_MAKE("usage_usec", "usage_usec"), vals3) == 1);
-        assert_se(vals3[0] && !vals3[1]);
-        free(vals3[0]);
+        assert(vals3[0] && !vals3[1]);
```
Comment (yuwata) [inline src/test/test-cgroup-util.c]: No no. Please use `assert_se()`. Please read my previous comment.

File: src/test/test-cgroup.c [hunk 7]
```diff
@@ -75,19 +76,19 @@ TEST(cg_create) {
 
         assert_se(cg_pid_get_path(SYSTEMD_CGROUP_CONTROLLER, getpid_cached(), &path) == 0);
         assert_se(streq(path, test_b));
-        free(path);
+        path = mfree(path);
```
Comment (yuwata) [inline src/test/test-cgroup.c]: Hm, this is not bad, but if you change this, then please change the declaration of `path` as `_cleanup_free_ char *path = NULL;`. This is much safer.

**Review by yuwata — CHANGES_REQUESTED** [review]

File: src/test/test-cgroup.c [hunk 8]
```diff
@@ -81,13 +82,13 @@ TEST(cg_create) {
...
 
         assert_se(cg_create_and_attach(SYSTEMD_CGROUP_CONTROLLER, test_d, 0) == 1);
 
         assert_se(cg_pid_get_path(SYSTEMD_CGROUP_CONTROLLER, getpid_cached(), &path) == 0);
         assert_se(path_equal(path, test_d));
-        free(path);
+        path = mfree(path);
 
         assert_se(cg_get_path(SYSTEMD_CGROUP_CONTROLLER, test_d, NULL, &path) == 0);
         log_debug("test_d: %s", path);
```
Comment (yuwata) [inline src/test/test-cgroup.c:94]: One more free() in the below must be replaced.

File: src/test/test-cgroup.c [hunk 9]
```diff
@@ -9,6 +9,7 @@
 #include "process-util.h"
 #include "string-util.h"
 #include "tests.h"
+#include "alloc-util.h"
```
Comment (yuwata) [inline src/test/test-cgroup.c:12]: Please sort headers.

**Review by yuwata — CHANGES_REQUESTED** [review]

**Comment by weizhixiangcoder** [comment]
好的，已经收到来自您的邮件！

**Comment by poettering** [comment]
Ping?

**Comment by weizhixiangcoder** [comment]
好的，已经收到来自您的邮件！

**Comment by poettering** [comment]
this seems to have stalled. let's close this. patch would certainly still be welcome if updated, with the raised issues addressed once it passes CI, but given it's stuck and this is mostly a clean-up, not a bugfix, let's just close this.

**Comment by weizhixiangcoder** [comment]
好的，已经收到来自您的邮件！

**Pull request closed without merging**